src/routes/classes/Classes.tsx

```diff
@@ -1,13 +1,43 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { RootState } from 'store/store';
+import { useSelector } from 'react-redux';
 import SectionHeader from 'components/SectionHeader/SectionHeader';
 import CardSkeleton from 'components/CardSkeleton/CardSkeleton';
+import JoinGroupForm from 'components/JoinGroupForm/JoinGroupForm';
+import CreateGroupForm from 'components/CreateGroupForm/CreateGroupForm';
+
+import { Subject } from 'types/Subject/Subject';
+import { getSubjects } from 'utils/db/db.utils';
 
 import styles from './Classes.module.css';
 
 export default function Classes() {
+  const user = useSelector((state: RootState) => state.user.currentUser);
+  const [classes, setClasses] = useState<Subject[]>([]);
+
+  useEffect(() => {
+    const getSubjectsList = async () => {
+      const data = await getSubjects(user?.authToken as string);
+
+      if (data.status === 'success') {
+        setClasses(data.data as Subject[]);
+      } else {
+        console.log(data.data);
```

Comment:
I am thinking what should happen if the request is not successful. I don't think a Toast would be our solution.
Maybe something like a retry button?

Reply:
I don't the retry button is the right approach. This fetch request is internal as it handles the list of classes that the teacher would see if he clicks the create a class button

Reply:
I have two approaches: We could add a loader in the class route, so that it fetches the data before loading the page. If there's an error, redirect the user to an error page. The other approach is that in the create group form, it there's no list, add a retry button.